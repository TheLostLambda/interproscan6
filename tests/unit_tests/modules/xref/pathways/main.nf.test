nextflow_process {

    name "Test Process PATHWAYS"
    script "interproscan/modules/xrefs/pathways/main.nf"
    process "PATHWAYS"

    stage {
        copy "tests/unit_tests/test_inputs/xrefs/pathways.json"
        copy "tests/unit_tests/test_inputs/xrefs/pathways.ipr.json"
        copy "interproscan/scripts/xrefs/pathways.py"
        copy "tests/unit_tests/test_inputs/xrefs/match_entries.json"
    }


    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                input[0] = file("meta/tests/unit_tests/test_inputs/xrefs/match_entries.json")
                input[1] = "../../../meta/tests/unit_tests/test_inputs/xrefs/pathways"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
