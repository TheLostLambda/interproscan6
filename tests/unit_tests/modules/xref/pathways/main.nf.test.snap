{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "pathways_match_entries.json:md5,688513254930e5a7b30a4d1b29a3b745"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-09-06T18:16:49.8948"
    }
}